pages/history.tsx

```diff
@@ -1,5 +1,135 @@
-export default function HistoryPage() {
+import {
+  Box,
+  Typography,
+  Button,
+  Unstable_Grid2 as Grid2,
+  useMediaQuery,
+  Autocomplete,
+  TextField,
+  Checkbox,
+  useTheme,
+} from '@mui/material'
+import SearchField from 'components/SearchField'
+import { GetServerSidePropsContext } from 'next'
+import { apiWrapper } from 'utils/apiWrappers'
+import { CategoryResponse, LogResponse } from 'utils/types'
+import { NextRouter, useRouter } from 'next/router'
+import categoriesHandler from '@api/categories'
+import { DatePicker } from '@mui/x-date-pickers'
+import HorizontalRuleIcon from '@mui/icons-material/HorizontalRule'
+import { addURLQueryParam, removeURLQueryParam } from 'utils/queryParams'
+
+export async function getServerSideProps(context: GetServerSidePropsContext) {
+  return {
+    props: { categories: await apiWrapper(categoriesHandler, context) },
+  }
+}
+
+interface HistoryPageProps {
+  logs: LogResponse[]
+  categories: CategoryResponse[]
+}
+
+const updateQuery = (router: NextRouter, key: string, val?: string) => {
+  if (!val) removeURLQueryParam(router, key)
+  else addURLQueryParam(router, key, val)
+}
+
+export default function HistoryPage({ logs, categories }: HistoryPageProps) {
+  const router = useRouter()
+  const theme = useTheme()
+  const isMobileView = useMediaQuery(theme.breakpoints.down('md'))
   return (
-    <h1>History</h1>
+    <Grid2 container my={2} sx={{ flexGrow: 1 }} gap={2}>
+      {/* Header -- "History" and "Export To Excel" button*/}
+      <Grid2 xs={12} container direction={'row'}>
+        <Typography variant="h5" sx={{ mb: 2, ml: 2 }}>
+          History
+        </Typography>
+        {!isMobileView && (
+          <Grid2 ml="auto" mr={6}>
+            <Button variant="outlined" sx={{ width: '100%' }}>
+              Export To Excel
+            </Button>
+          </Grid2>
+        )}
+      </Grid2>
+
+      {/* Filter views -- search, category dropdown, date range, internal only checkbox */}
+      <Grid2
+        container
+        xs={12}
+        direction={isMobileView ? 'column-reverse' : 'row'}
+        gap={isMobileView ? 2 : 4}
+        sx={{ px: 2, alignItems: 'center' }}
+      >
+        <Grid2 xs={12} md={2}>
+          <SearchField />
+        </Grid2>
+        <Grid2 xs={12} md={2}>
+          <Autocomplete
```

Comment:
use `SearchAutocomplete` component (this component does not have as many options as you have provided, take notes of what props we should add to it.)

Reply:
fixed